.github/workflows/clicky-workflow.yml

```diff
@@ -0,0 +1,47 @@
+name: "Clicky Download"
+on: 
+  schedule: 
+    - 
+      cron: "0 0 * * 0"
+  workflow_dispatch: ~
+
+jobs: 
+  build: 
+    runs-on: ubuntu-latest
+    steps: 
+      - 
+        name: "checkout branch"
+        uses: actions/checkout@v2
+      - 
+        name: "Setup python"
+        uses: actions/setup-python@v2
+        with: 
+          python-version: 3.8
+      - 
+        name: "Install dependencies"
+        run: |
+            python -m pip install --upgrade pip
+            pip install requests matplotlib
+      - 
+        name: "Run clicky script"
+        working-directory: "clicky"
+        run: |
+            python clickyScript.py
+      - 
+        name: Check dirty files
+        id: check
+        run: |
+          if git diff --exit-code; then
+            echo "::set-output name=diff::false"
+          else
+            echo "::set-output name=diff::true"
+          fi
+      - 
+        name: "Push changes"
+        if: steps.check.outputs.diff == 'true'
```

Comment:
Something seems to be not quite working in the logic of this conditional. This step of the workflow wasn't previously running, but removing this line allowed for the commit to successfully be made.

Reply:
I saw you removed this check, along with the check dirty files step.